### DISCUSSION TITLE
I cant import ZOAU api at ZOWE

### DISCUSSION THREAD

**Discussion by UallaceSantana** [IBM/zoau-samples#67] [Q&A]
Hi team,

I'm try to use zoautil_py at ZOWE in Vs Code. 

But I'm getting error, I dont know what I can do any more.

![image](https://github.com/IBM/zoau-samples/assets/88252393/13485218-691a-4800-b8d0-f1a7ebd90976)

![image](https://github.com/IBM/zoau-samples/assets/88252393/72a6019b-d12d-48bf-a9f0-f2e0930ffc4d)

The z/OS team has configured the ZOAU in these path: /global/tools/zoau/

How Can I import it correctly ?

**Comment by EnsonoStevenPerva** [comment]
vscode is indicating that it can't find the package local to your development environment. Are you able to execute the scripts on the remote host? I get these same errors, but the scripts execute just fine.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Uallace Santana ***@***.***>
Sent: Friday, May 26, 2023 9:48:06 AM
To: IBM/zoau-samples ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [IBM/zoau-samples] I cant import ZOAU api at ZOWE (Discussion #67)


*** ATTENTION!  This message originated from outside of Ensono. Treat hyperlinks and attachments in this email with caution. ***

THINK BEFORE YOU CLICK



Hi team,

I'm try to use zoautil_py at ZOWE in Vs Code.

But I'm getting error, I dont know what I can do any more.

[image]<https://user-images.githubusercontent.com/88252393/241251243-13485218-691a-4800-b8d0-f1a7ebd90976.png>

[image]<https://user-images.githubusercontent.com/88252393/241251272-72a6019b-d12d-48bf-a9f0-f2e0930ffc4d.png>

The z/OS team has configured the ZOAU in these path: /global/tools/zoau/

How Can I import it correctly ?

—
Reply to this email directly, view it on GitHub<https://github.com/IBM/zoau-samples/discussions/67>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AU3246WUVQN7XGOH3F3OKODXIC7CNANCNFSM6AAAAAAYQKV744>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

© 2021 Ensono, LP. All rights reserved. Ensono is a trademark of Ensono, LP. The information contained in this communication is confidential, is intended only for the use of the recipient named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution or copying of this communication is strictly prohibited. If you have received this communication in error, please resend this communication to the sender and delete the original message or any copy of it from your computer system.

**Comment by Degilio** [comment]
It might be cool to provide a zip file that can be expanded on the workstation and can resolve these things. I too just ignore them because we don’t have the references locally. I can’t think of any way to locally point to a remote location transparently for a PYTHONPATH


Frank J. De Gilio
CTO for Mainframe DeClunkification
IBM Systems Group

E-mail: ***@***.******@***.***>
Cell: 845-546-1130



From: Steven Perva (Ensono) ***@***.***>
Date: Friday, May 26, 2023 at 11:11
To: IBM/zoau-samples ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] Re: [IBM/zoau-samples] I cant import ZOAU api at ZOWE (Discussion #67)
vscode is indicating that it can't find the package local to your development environment. Are you able to execute the scripts on the remote host? I get these same errors, but the scripts execute just fine. Get Outlook for iOS<https: //aka. ms/o0ukef>
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd
vscode is indicating that it can't find the package local to your development environment. Are you able to execute the scripts on the remote host? I get these same errors, but the scripts execute just fine.

Get Outlook for iOS<https://aka.ms/o0ukef><https://aka.ms/o0ukef%3E>
________________________________
From: Uallace Santana ***@***.***>
Sent: Friday, May 26, 2023 9:48:06 AM
To: IBM/zoau-samples ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [IBM/zoau-samples] I cant import ZOAU api at ZOWE (Discussion #67)


*** ATTENTION! This message originated from outside of Ensono. Treat hyperlinks and attachments in this email with caution. ***

THINK BEFORE YOU CLICK



Hi team,

I'm try to use zoautil_py at ZOWE in Vs Code.

But I'm getting error, I dont know what I can do any more.

[image]<https://user-images.githubusercontent.com/88252393/241251243-13485218-691a-4800-b8d0-f1a7ebd90976.png><https://user-images.githubusercontent.com/88252393/241251243-13485218-691a-4800-b8d0-f1a7ebd90976.png%3E>

[image]<https://user-images.githubusercontent.com/88252393/241251272-72a6019b-d12d-48bf-a9f0-f2e0930ffc4d.png><https://user-images.githubusercontent.com/88252393/241251272-72a6019b-d12d-48bf-a9f0-f2e0930ffc4d.png%3E>

The z/OS team has configured the ZOAU in these path: /global/tools/zoau/

How Can I import it correctly ?

—
Reply to this email directly, view it on GitHub<https://github.com/IBM/zoau-samples/discussions/67><https://github.com/IBM/zoau-samples/discussions/67%3E>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AU3246WUVQN7XGOH3F3OKODXIC7CNANCNFSM6AAAAAAYQKV744><https://github.com/notifications/unsubscribe-auth/AU3246WUVQN7XGOH3F3OKODXIC7CNANCNFSM6AAAAAAYQKV744%3E>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

© 2021 Ensono, LP. All rights reserved. Ensono is a trademark of Ensono, LP. The information contained in this communication is confidential, is intended only for the use of the recipient named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution or copying of this communication is strictly prohibited. If you have received this communication in error, please resend this communication to the sender and delete the original message or any copy of it from your computer system.

**Comment by FernandoLRubio** [comment]
Hi there! We're currently working on stub `.pyi` files for vscode. This will allow both autocompletion and syntax analysis. In which OS are you running vscode?

**Reply by aviVzm05** [reply]
from the https://www.ibm.com/docs/en/zoau/1.2.x?topic=whats-new-noteworthy site, I understand we now have zoautil_py-stubs, can you let us know,  how  or from where we can download these and any instructions on setup would be helpful.

**Reply by EnsonoStevenPerva** [reply]
@aviVzm05 your stub files should be located on your z/OS LPAR, wherever `ZOAU_HOME` is (in my case `/usr/lpp/IBM/zoautil`) and from there look for the `zoautil_py-stubs` directory. You should download this entire directory.

You can do that with FTP or SFTP (definitely recommend secure FTP when possible). Once you have the whole directory downloaded to your workstation, you'll need to follow your IDEs directions for placing stubs where they need to go for the development environment to use them. The directions will vary from IDE to IDE.

In vscode, I saved the stubs to a directory of my choosing (`C:\Development\Python\Stubs\`) and edited my `settings.json` and added a line that said `    "python.analysis.stubPath": "C:\\Development\\Python\\Stubs",` and it all worked. There's probably a much easier way but I don't know what it is.

**Reply by aviVzm05** [reply]
Thank you, we are still in ZOAU version 1.2.3. Guess we need to upgrade to 1.2.4 for this.
